iib/workers/tasks/build_merge_index_image.py

```diff
@@ -128,9 +128,10 @@ def _add_bundles_missing_in_source(
             missing_bundles.append(bundle)
             missing_bundle_paths.append(bundle['bundlePath'])
 
-    if ignore_bundle_ocp_version and target_index is not None:
+    if ignore_bundle_ocp_version:
+        target_index_tmp = '' if target_index is None else target_index
         allow_no_ocp_version = any(
-            target_index.startswith(index)
+            target_index_tmp.startswith(index) or source_from_index.startswith(index)
             for index in get_worker_config()['iib_no_ocp_label_allow_list']
```

Comment:
Does this change take into account that "random_text".startswith('') will result in True?

Reply:
well, it did not . However I added config validation to ensure nobody sets empty string in the allow list.
Please re-review

Reply:
My bad, when I was thinking about this, I switched the parameters (I thought the target index was checked in `startswith()` method). However, this check is useful in my opinion.